File: backend/verify-dashboard-data.mjs
/**
 * Dashboard Data Verification Script
 *
 * Comprehensive check of all dashboard data quality
 */

import { createClient } from '@supabase/supabase-js';

const SUPABASE_URL = 'https://lbjymlodxprzqgtyqtcq.supabase.co';
const SUPABASE_SERVICE_ROLE_KEY = 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6ImxianltbG9keHByenFndHlxdGNxIiwicm9sZSI6InNlcnZpY2Vfcm9sZSIsImlhdCI6MTc2MzM0OTUzMSwiZXhwIjoyMDc4OTI1NTMxfQ.5B2bBOUgXw2qC6QpIkU2DNZiqMW20y8MzgLQ56LEtFA';

const supabase = createClient(SUPABASE_URL, SUPABASE_SERVICE_ROLE_KEY, {
  auth: {
    persistSession: false,
    autoRefreshToken: false,
  },
});

const results = [];

async function logResult(check, status, message, details) {
  results.push({ check, status, message, details });
  const emoji = status === 'PASS' ? '✅' : status === 'FAIL' ? '❌' : '⚠️ ';
  console.log(`${emoji} ${status}: ${check} - ${message}`);
  if (details) {
    console.log('   Details:', JSON.stringify(details, null, 2));
  }
}

// Check 1: Database Connectivity
async function checkDatabaseConnectivity() {
  try {
    const { data, error } = await supabase
      .from('calls')
      .select('count')
      .limit(1);

    if (error) {
      await logResult('Database Connectivity', 'FAIL', `Cannot connect to database: ${error.message}`);
    } else {
      await logResult('Database Connectivity', 'PASS', 'Successfully connected to Supabase');
    }
  } catch (err) {
    await logResult('Database Connectivity', 'FAIL', `Connection error: ${err.message}`);
  }
}

// Check 2: Caller Names
async function checkCallerNames() {
  try {
    const sevenDaysAgo = new Date(Date.now() - 7 * 24 * 60 * 60 * 1000).toISOString();
    const { data: callsData, error } = await supabase
      .from('calls')
      .select('caller_name')
      .gte('created_at', sevenDaysAgo);

    if (error) {
      await logResult('Caller Names', 'FAIL', `Database query failed: ${error.message}`);
      return;
    }

    const totalCalls = callsData.length;
    const callsWithNames = callsData.filter(c => c.caller_name && c.caller_name !== 'Unknown Caller').length;
    const unknownCallers = totalCalls - callsWithNames;

    const details = { total_calls: totalCalls, calls_with_names: callsWithNames, unknown_callers: unknownCallers };

    if (totalCalls === 0) {
      await logResult('Caller Names', 'WARN', 'No calls in the last 7 days', details);
    } else if (callsWithNames === 0) {
      await logResult('Caller Names', 'FAIL', 'ALL calls are "Unknown Caller" - enrichment broken', details);
    } else if (callsWithNames / totalCalls < 0.5) {
      await logResult('Caller Names', 'WARN', `Low enrichment rate: ${Math.round(callsWithNames / totalCalls * 100)}%`, details);
    } else {
      await logResult('Caller Names', 'PASS', `${callsWithNames}/${totalCalls} calls have enriched names`, details);
    }
  } catch (err) {
    await logResult('Caller Names', 'FAIL', `Error: ${err.message}`);
  }
}

// Check 3: Contact Names
async function checkContactNames() {
  try {
    const sevenDaysAgo = new Date(Date.now() - 7 * 24 * 60 * 60 * 1000).toISOString();
    const { data: contactsData, error } = await supabase
      .from('contacts')
      .select('name')
      .gte('created_at', sevenDaysAgo);

    if (error) {
      await logResult('Contact Names', 'FAIL', `Database query failed: ${error.message}`);
      return;
    }

    const totalContacts = contactsData.length;
    const contactsWithNames = contactsData.filter(c => c.name && c.name !== 'Unknown Caller').length;

    const details = { total_contacts: totalContacts, contacts_with_names: contactsWithNames };

    if (totalContacts === 0) {
      await logResult('Contact Names', 'WARN', 'No contacts in the last 7 days', details);
    } else if (contactsWithNames === 0) {
      await logResult('Contact Names', 'FAIL', 'ALL contacts have no names', details);
    } else {
      await logResult('Contact Names', 'PASS', `${contactsWithNames}/${totalContacts} contacts have names`, details);
    }
  } catch (err) {
    await logResult('Contact Names', 'FAIL', `Error: ${err.message}`);
  }
}

// Check 4: Sentiment Data
async function checkSentimentData() {
  try {
    const sevenDaysAgo = new Date(Date.now() - 7 * 24 * 60 * 60 * 1000).toISOString();
    const { data: callsData, error } = await supabase
      .from('calls')
      .select('sentiment_label, sentiment_score')
      .gte('created_at', sevenDaysAgo);

    if (error) {
      await logResult('Sentiment Data', 'FAIL', `Database query failed: ${error.message}`);
      return;
    }

    const totalCalls = callsData.length;
    const callsWithSentimentLabel = callsData.filter(c => c.sentiment_label).length;
    const callsWithSentimentScore = callsData.filter(c => c.sentiment_score !== null && c.sentiment_score !== undefined).length;

    const scores = callsData.filter(c => c.sentiment_score !== null).map(c => c.sentiment_score);
    const avgSentimentScore = scores.length > 0 ? scores.reduce((a, b) => a + b, 0) / scores.length : 0;

    const details = {
      total_calls: totalCalls,
      calls_with_sentiment_label: callsWithSentimentLabel,
      calls_with_sentiment_score: callsWithSentimentScore,
      avg_sentiment_score: avgSentimentScore.toFixed(2),
    };

    if (totalCalls === 0) {
      await logResult('Sentiment Data', 'WARN', 'No calls in the last 7 days', details);
    } else if (avgSentimentScore === 0 && totalCalls > 0) {
      await logResult('Sentiment Data', 'FAIL', 'Average sentiment is 0% - sentiment analysis BROKEN', details);
    } else if (callsWithSentimentScore === 0) {
      await logResult('Sentiment Data', 'FAIL', 'No calls have sentiment scores', details);
    } else {
      await logResult('Sentiment Data', 'PASS', `Avg sentiment: ${(avgSentimentScore * 100).toFixed(1)}%, ${callsWithSentimentScore}/${totalCalls} calls scored`, details);
    }
  } catch (err) {
    await logResult('Sentiment Data', 'FAIL', `Error: ${err.message}`);
  }
}

// Check 5: Hot Lead Alerts
async function checkHotLeadAlerts() {
  try {
    const sevenDaysAgo = new Date(Date.now() - 7 * 24 * 60 * 60 * 1000).toISOString();
    const { data: alertsData, error } = await supabase
      .from('hot_lead_alerts')
      .select('lead_score')
      .gte('created_at', sevenDaysAgo);

    if (error) {
      await logResult('Hot Lead Alerts', 'FAIL', `Database query failed: ${error.message}`);
      return;
    }

    const totalAlerts = alertsData.length;
    const alertsAboveThreshold = alertsData.filter(a => a.lead_score >= 60).length;

    const details = { total_alerts: totalAlerts, alerts_above_threshold: alertsAboveThreshold };

    if (totalAlerts === 0) {
      await logResult('Hot Lead Alerts', 'WARN', 'No hot lead alerts in the last 7 days', details);
    } else {
      await logResult('Hot Lead Alerts', 'PASS', `${totalAlerts} alerts found (${alertsAboveThreshold} above threshold)`, details);
    }
  } catch (err) {
    await logResult('Hot Lead Alerts', 'FAIL', `Error: ${err.message}`);
  }
}

// Check 6: Phone Numbers
async function checkPhoneNumbers() {
  try {
    const sevenDaysAgo = new Date(Date.now() - 7 * 24 * 60 * 60 * 1000).toISOString();
    const { data: callsData, error } = await supabase
      .from('calls')
      .select('phone_number')
      .gte('created_at', sevenDaysAgo);

    if (error) {
      await logResult('Phone Numbers', 'FAIL', `Database query failed: ${error.message}`);
      return;
    }

    const total = callsData.length;
    const e164Format = callsData.filter(c => c.phone_number && c.phone_number.startsWith('+')).length;

    const details = { total, e164_format: e164Format };

    if (total === 0) {
      await logResult('Phone Numbers', 'WARN', 'No calls in the last 7 days', details);
    } else if (e164Format !== total) {
      await logResult('Phone Numbers', 'FAIL', `${total - e164Format}/${total} phone numbers NOT in E.164 format`, details);
    } else {
      await logResult('Phone Numbers', 'PASS', `All ${total} phone numbers in E.164 format`, details);
    }
  } catch (err) {
    await logResult('Phone Numbers', 'FAIL', `Error: ${err.message}`);
  }
}

// Check 7: Sample Data
async function checkSampleData() {
  try {
    const { data: sampleData, error } = await supabase
      .from('calls')
      .select('id, caller_name, phone_number, sentiment_label, sentiment_score, call_direction, status, created_at')
      .order('created_at', { ascending: false })
      .limit(5);

    if (error) {
      await logResult('Sample Data', 'FAIL', `Database query failed: ${error.message}`);
      return;
    }

    if (!sampleData || sampleData.length === 0) {
      await logResult('Sample Data', 'WARN', 'No calls found in database', {});
    } else {
      await logResult('Sample Data', 'PASS', `Retrieved ${sampleData.length} recent calls`, sampleData);
    }
  } catch (err) {
    await logResult('Sample Data', 'FAIL', `Error: ${err.message}`);
  }
}

// Check 8: Total Call Volume
async function checkTotalCallVolume() {
  try {
    const { count, error } = await supabase
      .from('calls')
      .select('*', { count: 'exact', head: true });

    if (error) {
      await logResult('Total Call Volume', 'FAIL', `Cannot count calls: ${error.message}`);
    } else {
      await logResult('Total Call Volume', 'PASS', `Total calls in database: ${count}`, { count });
    }
  } catch (err) {
    await logResult('Total Call Volume', 'FAIL', `Error: ${err.message}`);
  }
}

// Check 9: Recent Call Activity
async function checkRecentCallActivity() {
  try {
    const { data, error } = await supabase
      .from('calls')
      .select('created_at')
      .order('created_at', { ascending: false })
      .limit(1);

    if (error) {
      await logResult('Recent Call Activity', 'FAIL', `Cannot query recent calls: ${error.message}`);
      return;
    }

    if (!data || data.length === 0) {
      await logResult('Recent Call Activity', 'WARN', 'No calls found in database', {});
    } else {
      const latestCall = new Date(data[0].created_at);
      const hoursSinceLastCall = (Date.now() - latestCall.getTime()) / (1000 * 60 * 60);

      if (hoursSinceLastCall > 24) {
        await logResult('Recent Call Activity', 'WARN', `Last call was ${Math.round(hoursSinceLastCall)} hours ago`, { latest_call: latestCall.toISOString() });
      } else {
        await logResult('Recent Call Activity', 'PASS', `Last call was ${Math.round(hoursSinceLastCall)} hours ago`, { latest_call: latestCall.toISOString() });
      }
    }
  } catch (err) {
    await logResult('Recent Call Activity', 'FAIL', `Error: ${err.message}`);
  }
}

// Main execution
async function main() {
  console.log('========================================');
  console.log('DASHBOARD DATA VERIFICATION');
  console.log('========================================\n');

  console.log('Starting comprehensive verification...\n');

  // Run all checks
  await checkDatabaseConnectivity();
  await checkTotalCallVolume();
  await checkRecentCallActivity();
  await checkCallerNames();
  await checkContactNames();
  await checkSentimentData();
  await checkHotLeadAlerts();
  await checkPhoneNumbers();
  await checkSampleData();

  // Summary
  console.log('\n========================================');
  console.log('VERIFICATION SUMMARY');
  console.log('========================================\n');

  const passed = results.filter(r => r.status === 'PASS').length;
  const failed = results.filter(r => r.status === 'FAIL').length;
  const warnings = results.filter(r => r.status === 'WARN').length;

  console.log(`✅ Passed: ${passed}/${results.length} checks`);
  console.log(`❌ Failed: ${failed}/${results.length} checks`);
  console.log(`⚠️  Warnings: ${warnings}/${results.length} checks\n`);

  if (failed > 0) {
    console.log('CRITICAL ISSUES:');
    results.filter(r => r.status === 'FAIL').forEach(r => {
      console.log(`  ❌ ${r.check}: ${r.message}`);
    });
    console.log('');
  }

  if (warnings > 0) {
    console.log('WARNINGS:');
    results.filter(r => r.status === 'WARN').forEach(r => {
      console.log(`  ⚠️  ${r.check}: ${r.message}`);
    });
    console.log('');
  }

  // Sample data display
  const sampleResult = results.find(r => r.check === 'Sample Data');
  if (sampleResult && sampleResult.details && Array.isArray(sampleResult.details)) {
    console.log('\nSAMPLE DATA (5 most recent calls):');
    console.log('====================================');
    sampleResult.details.forEach((call, idx) => {
      console.log(`\nCall ${idx + 1}:`);
      console.log(`  ID: ${call.id}`);
      console.log(`  Caller: ${call.caller_name || 'N/A'}`);
      console.log(`  Phone: ${call.phone_number || 'N/A'}`);
      console.log(`  Sentiment: ${call.sentiment_label || 'N/A'} (${call.sentiment_score || 0})`);
      console.log(`  Direction: ${call.call_direction || 'N/A'}`);
      console.log(`  Status: ${call.status || 'N/A'}`);
      console.log(`  Created: ${call.created_at || 'N/A'}`);
    });
  }

  console.log('\n========================================');
  console.log('VERIFICATION COMPLETE');
  console.log('========================================\n');

  // Exit with appropriate code
  process.exit(failed > 0 ? 1 : 0);
}

main().catch(error => {
  console.error('❌ FATAL ERROR:', error.message);
  console.error(error);
  process.exit(1);
});

File: verify-dashboard-data.py
#!/usr/bin/env python3
"""
Dashboard Data Verification Script

Comprehensive check of all dashboard data quality using direct PostgreSQL connection
"""

import psycopg2
from datetime import datetime, timedelta
import json
import sys

# Database connection string
DATABASE_URL = "postgresql://postgres:Eguale%402021%3F@db.lbjymlodxprzqgtyqtcq.supabase.co:5432/postgres"

results = []

def log_result(check, status, message, details=None):
    """Log a verification result"""
    results.append({
        'check': check,
        'status': status,
        'message': message,
        'details': details
    })
    emoji = '✅' if status == 'PASS' else '❌' if status == 'FAIL' else '⚠️ '
    print(f"{emoji} {status}: {check} - {message}")
    if details:
        print(f"   Details: {json.dumps(details, indent=2)}")

def check_database_connectivity(conn):
    """Check if database connection works"""
    try:
        cursor = conn.cursor()
        cursor.execute("SELECT 1")
        cursor.fetchone()
        log_result('Database Connectivity', 'PASS', 'Successfully connected to Supabase')
        cursor.close()
    except Exception as e:
        log_result('Database Connectivity', 'FAIL', f'Cannot connect to database: {str(e)}')

def check_caller_names(conn):
    """Check caller name enrichment"""
    try:
        cursor = conn.cursor()
        seven_days_ago = (datetime.now() - timedelta(days=7)).isoformat()

        cursor.execute("""
            SELECT
                COUNT(*) as total_calls,
                COUNT(CASE WHEN caller_name IS NOT NULL AND caller_name != 'Unknown Caller' THEN 1 END) as calls_with_names,
                COUNT(CASE WHEN caller_name = 'Unknown Caller' OR caller_name IS NULL THEN 1 END) as unknown_callers
            FROM calls
            WHERE created_at > %s
        """, (seven_days_ago,))

        result = cursor.fetchone()
        total_calls = result[0]
        calls_with_names = result[1]
        unknown_callers = result[2]

        details = {
            'total_calls': total_calls,
            'calls_with_names': calls_with_names,
            'unknown_callers': unknown_callers
        }

        if total_calls == 0:
            log_result('Caller Names', 'WARN', 'No calls in the last 7 days', details)
        elif calls_with_names == 0:
            log_result('Caller Names', 'FAIL', 'ALL calls are "Unknown Caller" - enrichment broken', details)
        elif calls_with_names / total_calls < 0.5:
            log_result('Caller Names', 'WARN', f'Low enrichment rate: {round(calls_with_names / total_calls * 100)}%', details)
        else:
            log_result('Caller Names', 'PASS', f'{calls_with_names}/{total_calls} calls have enriched names', details)

        cursor.close()
    except Exception as e:
        log_result('Caller Names', 'FAIL', f'Database query failed: {str(e)}')

def check_contact_names(conn):
    """Check contact name enrichment"""
    try:
        cursor = conn.cursor()
        seven_days_ago = (datetime.now() - timedelta(days=7)).isoformat()

        cursor.execute("""
            SELECT
                COUNT(*) as total_contacts,
                COUNT(CASE WHEN name IS NOT NULL AND name != 'Unknown Caller' THEN 1 END) as contacts_with_names
            FROM contacts
            WHERE created_at > %s
        """, (seven_days_ago,))

        result = cursor.fetchone()
        total_contacts = result[0]
        contacts_with_names = result[1]

        details = {
            'total_contacts': total_contacts,
            'contacts_with_names': contacts_with_names
        }

        if total_contacts == 0:
            log_result('Contact Names', 'WARN', 'No contacts in the last 7 days', details)
        elif contacts_with_names == 0:
            log_result('Contact Names', 'FAIL', 'ALL contacts have no names', details)
        else:
            log_result('Contact Names', 'PASS', f'{contacts_with_names}/{total_contacts} contacts have names', details)

        cursor.close()
    except Exception as e:
        log_result('Contact Names', 'FAIL', f'Database query failed: {str(e)}')

def check_sentiment_data(conn):
    """Check sentiment analysis data"""
    try:
        cursor = conn.cursor()
        seven_days_ago = (datetime.now() - timedelta(days=7)).isoformat()

        cursor.execute("""
            SELECT
                COUNT(*) as total_calls,
                COUNT(sentiment_label) as calls_with_sentiment_label,
                COUNT(sentiment_score) as calls_with_sentiment_score,
                AVG(sentiment_score) as avg_sentiment_score
            FROM calls
            WHERE created_at > %s
        """, (seven_days_ago,))

        result = cursor.fetchone()
        total_calls = result[0]
        calls_with_sentiment_label = result[1]
        calls_with_sentiment_score = result[2]
        avg_sentiment_score = result[3] or 0

        details = {
            'total_calls': total_calls,
            'calls_with_sentiment_label': calls_with_sentiment_label,
            'calls_with_sentiment_score': calls_with_sentiment_score,
            'avg_sentiment_score': f'{avg_sentiment_score:.2f}'
        }

        if total_calls == 0:
            log_result('Sentiment Data', 'WARN', 'No calls in the last 7 days', details)
        elif avg_sentiment_score == 0 and total_calls > 0:
            log_result('Sentiment Data', 'FAIL', 'Average sentiment is 0% - sentiment analysis BROKEN', details)
        elif calls_with_sentiment_score == 0:
            log_result('Sentiment Data', 'FAIL', 'No calls have sentiment scores', details)
        else:
            log_result('Sentiment Data', 'PASS', f'Avg sentiment: {avg_sentiment_score * 100:.1f}%, {calls_with_sentiment_score}/{total_calls} calls scored', details)

        cursor.close()
    except Exception as e:
        log_result('Sentiment Data', 'FAIL', f'Database query failed: {str(e)}')

def check_hot_lead_alerts(conn):
    """Check hot lead alerts"""
    try:
        cursor = conn.cursor()
        seven_days_ago = (datetime.now() - timedelta(days=7)).isoformat()

        cursor.execute("""
            SELECT
                COUNT(*) as total_alerts,
                COUNT(CASE WHEN lead_score >= 60 THEN 1 END) as alerts_above_threshold
            FROM hot_lead_alerts
            WHERE created_at > %s
        """, (seven_days_ago,))

        result = cursor.fetchone()
        total_alerts = result[0]
        alerts_above_threshold = result[1]

        details = {
            'total_alerts': total_alerts,
            'alerts_above_threshold': alerts_above_threshold
        }

        if total_alerts == 0:
            log_result('Hot Lead Alerts', 'WARN', 'No hot lead alerts in the last 7 days', details)
        else:
            log_result('Hot Lead Alerts', 'PASS', f'{total_alerts} alerts found ({alerts_above_threshold} above threshold)', details)

        cursor.close()
    except Exception as e:
        log_result('Hot Lead Alerts', 'FAIL', f'Database query failed: {str(e)}')

def check_phone_numbers(conn):
    """Check phone number formatting"""
    try:
        cursor = conn.cursor()
        seven_days_ago = (datetime.now() - timedelta(days=7)).isoformat()

        cursor.execute("""
            SELECT
                COUNT(*) as total,
                COUNT(CASE WHEN phone_number LIKE '+%%' THEN 1 END) as e164_format
            FROM calls
            WHERE created_at > %s
        """, (seven_days_ago,))

        result = cursor.fetchone()
        total = result[0]
        e164_format = result[1]

        details = {
            'total': total,
            'e164_format': e164_format
        }

        if total == 0:
            log_result('Phone Numbers', 'WARN', 'No calls in the last 7 days', details)
        elif e164_format != total:
            log_result('Phone Numbers', 'FAIL', f'{total - e164_format}/{total} phone numbers NOT in E.164 format', details)
        else:
            log_result('Phone Numbers', 'PASS', f'All {total} phone numbers in E.164 format', details)

        cursor.close()
    except Exception as e:
        log_result('Phone Numbers', 'FAIL', f'Database query failed: {str(e)}')

def check_sample_data(conn):
    """Check sample recent calls"""
    try:
        cursor = conn.cursor()

        cursor.execute("""
            SELECT
                id,
                caller_name,
                phone_number,
                sentiment_label,
                sentiment_score,
                call_direction,
                status,
                created_at
            FROM calls
            ORDER BY created_at DESC
            LIMIT 5
        """)

        rows = cursor.fetchall()

        if not rows:
            log_result('Sample Data', 'WARN', 'No calls found in database', {})
        else:
            sample_data = []
            for row in rows:
                sample_data.append({
                    'id': row[0],
                    'caller_name': row[1],
                    'phone_number': row[2],
                    'sentiment_label': row[3],
                    'sentiment_score': row[4],
                    'call_direction': row[5],
                    'status': row[6],
                    'created_at': str(row[7])
                })
            log_result('Sample Data', 'PASS', f'Retrieved {len(sample_data)} recent calls', sample_data)

        cursor.close()
    except Exception as e:
        log_result('Sample Data', 'FAIL', f'Database query failed: {str(e)}')

def check_total_call_volume(conn):
    """Check total call volume"""
    try:
        cursor = conn.cursor()
        cursor.execute("SELECT COUNT(*) FROM calls")
        count = cursor.fetchone()[0]
        log_result('Total Call Volume', 'PASS', f'Total calls in database: {count}', {'count': count})
        cursor.close()
    except Exception as e:
        log_result('Total Call Volume', 'FAIL', f'Cannot count calls: {str(e)}')

def check_recent_call_activity(conn):
    """Check recent call activity"""
    try:
        cursor = conn.cursor()
        cursor.execute("SELECT created_at FROM calls ORDER BY created_at DESC LIMIT 1")
        result = cursor.fetchone()

        if not result:
            log_result('Recent Call Activity', 'WARN', 'No calls found in database', {})
        else:
            latest_call = result[0]
            hours_since_last_call = (datetime.now() - latest_call).total_seconds() / 3600

            details = {'latest_call': str(latest_call)}

            if hours_since_last_call > 24:
                log_result('Recent Call Activity', 'WARN', f'Last call was {round(hours_since_last_call)} hours ago', details)
            else:
                log_result('Recent Call Activity', 'PASS', f'Last call was {round(hours_since_last_call)} hours ago', details)

        cursor.close()
    except Exception as e:
        log_result('Recent Call Activity', 'FAIL', f'Cannot query recent calls: {str(e)}')

def main():
    """Main execution"""
    print('=' * 40)
    print('DASHBOARD DATA VERIFICATION')
    print('=' * 40)
    print()
    print('Starting comprehensive verification...')
    print()

    try:
        # Connect to database
        conn = psycopg2.connect(DATABASE_URL)

        # Run all checks
        check_database_connectivity(conn)
        check_total_call_volume(conn)
        check_recent_call_activity(conn)
        check_caller_names(conn)
        check_contact_names(conn)
        check_sentiment_data(conn)
        check_hot_lead_alerts(conn)
        check_phone_numbers(conn)
        check_sample_data(conn)

        # Close connection
        conn.close()

    except Exception as e:
        print(f"❌ FATAL ERROR: Failed to connect to database: {str(e)}")
        sys.exit(1)

    # Summary
    print()
    print('=' * 40)
    print('VERIFICATION SUMMARY')
    print('=' * 40)
    print()

    passed = len([r for r in results if r['status'] == 'PASS'])
    failed = len([r for r in results if r['status'] == 'FAIL'])
    warnings = len([r for r in results if r['status'] == 'WARN'])

    print(f"✅ Passed: {passed}/{len(results)} checks")
    print(f"❌ Failed: {failed}/{len(results)} checks")
    print(f"⚠️  Warnings: {warnings}/{len(results)} checks")
    print()

    if failed > 0:
        print('CRITICAL ISSUES:')
        for r in results:
            if r['status'] == 'FAIL':
                print(f"  ❌ {r['check']}: {r['message']}")
        print()

    if warnings > 0:
        print('WARNINGS:')
        for r in results:
            if r['status'] == 'WARN':
                print(f"  ⚠️  {r['check']}: {r['message']}")
        print()

    # Sample data display
    sample_result = next((r for r in results if r['check'] == 'Sample Data'), None)
    if sample_result and sample_result['details'] and isinstance(sample_result['details'], list):
        print('\nSAMPLE DATA (5 most recent calls):')
        print('=' * 40)
        for idx, call in enumerate(sample_result['details'], 1):
            print(f"\nCall {idx}:")
            print(f"  ID: {call.get('id', 'N/A')}")
            print(f"  Caller: {call.get('caller_name', 'N/A')}")
            print(f"  Phone: {call.get('phone_number', 'N/A')}")
            print(f"  Sentiment: {call.get('sentiment_label', 'N/A')} ({call.get('sentiment_score', 0)})")
            print(f"  Direction: {call.get('call_direction', 'N/A')}")
            print(f"  Status: {call.get('status', 'N/A')}")
            print(f"  Created: {call.get('created_at', 'N/A')}")

    print()
    print('=' * 40)
    print('VERIFICATION COMPLETE')
    print('=' * 40)
    print()

    # Exit with appropriate code
    sys.exit(1 if failed > 0 else 0)

if __name__ == '__main__':
    main()
